### PR TITLE
Individual Event Subscription-8642 

### PR DESCRIPTION
**Pull request by sericcson**

### CODE REVIEW

File: src/main/java/com/ericsson/ei/controller/RuleCheckControllerImpl.java [hunk 1]
```diff
@@ -46,12 +46,20 @@
...
      * @param rule-
      *            takes a String as a rule that need to be evaluated on JSON
      *            content
+=======
+     * This method interacts with JmesPathInterface class method runRuleOnEvent to
+     * evaluate a rule on JSON object.
+     * 
+     * @param rule-
+     *            takes a String as a rule that need to be evaluated on JSON content
+>>>>>>> a516e969eb13fedb1741cf8c4f2d76d776655f23
```
Comment (tobiasake) [inline src/main/java/com/ericsson/ei/controller/RuleCheckControllerImpl.java]: Handle/fix/remove Git merge comments.

File: src/test/java/com/ericsson/ei/flowtests/SingleEventAggregationTest.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            for (String temp : eventNames) {
+                String expectedEvent = eventsJson.at("/" + temp).toString();
+                String actualEventID = eventsJson.at("/" + temp + "/meta/id").asText();
+                String document = objectHandler.findObjectById(actualEventID);
+                JsonNode actualJson = objectmapper.readTree(document);
+                JsonNode actualEvent = actualJson.at("/aggregatedObject");
+                ObjectNode objectActualEvent = (ObjectNode) actualEvent;
+                objectActualEvent.without("TemplateName");
+                assertEquals(objectActualEvent.toString(), expectedEvent);
+            }
```
Comment (tobiasake) [inline src/test/java/com/ericsson/ei/flowtests/SingleEventAggregationTest.java]: Is it possible to get any test report of which events is missing and so on, if it fails.
A test report that says what failed and which event was missing during the failed test execution should really be helpful when debugging.

File: pom.xml [hunk 3]
```diff
@@ -3,10 +3,12 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.github.ericsson</groupId>
     <artifactId>eiffel-intelligence</artifactId>
+
```
Comment (evasiba-ericsson) [inline pom.xml]: why this empty line

File: pom.xml [hunk 4]
```diff
@@ -3,10 +3,12 @@
...
     <version>0.0.4</version>
     
     <packaging>war</packaging>
 
+
```
Comment (evasiba-ericsson) [inline pom.xml]: why this empty line

File: src/test/resources/SubscriptionRules.json [hunk 5]
```diff
@@ -0,0 +1,577 @@
+[
```
Comment (evasiba-ericsson) [inline src/test/resources/SubscriptionRules.json]: Change name of this file to all_events_rules.json

File: src/test/resources/SubscriptionObjectForEmailTest.json [hunk 6]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (evasiba-ericsson) [inline src/test/resources/SubscriptionObjectForEmailTest.json]: What is this file doing in this pull request?

File: src/main/resources/SubscriptionRules.json [hunk 7]
```diff
@@ -0,0 +1,577 @@
+[
```
Comment (evasiba-ericsson) [inline src/main/resources/SubscriptionRules.json]: change name of this file to all_event_rules.json

File: src/test/java/com/ericsson/ei/flowtests/SingleEventAggregationTest.java [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+        } catch (Exception e) {
+            log.info(e.getMessage(), e);
```
Comment (evasiba-ericsson) [inline src/test/java/com/ericsson/ei/flowtests/SingleEventAggregationTest.java]: change to log.error

File: src/test/java/com/ericsson/ei/flowtests/SingleEventAggregationTest.java (hunk 2 above)
Comment (tobiasake) [reply src/test/java/com/ericsson/ei/flowtests/SingleEventAggregationTest.java]: I have created separate story for improving the test result report of what have failed.

**Pull request closed without merging**